### ISSUE TITLE
Pagination bug with total count for the latest page

### ISSUE THREAD

**Issue by ruscoder** [Aidbox/Issues#103]
devbox:0.4.4-SNAPSHOT

Sometimes I get this strange behaviour

```
resourceType: Bundle
type: searchset
entry:
- resource: {id: c7200021-115f-48ae-b392-7eabc524f2f6, resourceType: Patient}
  fullUrl: http://devbox:8080/Patient/c7200021-115f-48ae-b392-7eabc524f2f6
... 8 entries here...
- resource: {id: 38e43ab0-d205-4972-9fdf-1145b02fc323, resourceType: Patient}
  fullUrl: http://devbox:8080/Patient/38e43ab0-d205-4972-9fdf-1145b02fc323
total: 10
link:
- {relation: first, url: '/Patient?_page=1&_count=15&_elements=id'}
- {relation: self, url: '/Patient?_page=1529&_count=15&_elements=id'}
query-sql: ['SELECT "patient".* FROM "patient" LIMIT ? OFFSET ? ', 15, 22920]
query-time: 10
```

You can see that Aidbox returned 10 entries (and the total is 10) but for page 1529 I have at least 22920 entries.

But for the previous page everything is OK:
```
resourceType: Bundle
type: searchset
entry:
- resource: {id: b78094c3-468f-417f-badc-60f78721fd46, resourceType: Patient}
  fullUrl: http://devbox:8080/Patient/b78094c3-468f-417f-badc-60f78721fd46
... 13 entries here...
- resource: {id: c0e95dce-50e0-4b7a-8692-6cfcbaf88cc1, resourceType: Patient}
  fullUrl: http://devbox:8080/Patient/c0e95dce-50e0-4b7a-8692-6cfcbaf88cc1
total: 22930
link:
- {relation: first, url: '/Patient?_page=1&_count=15&_elements=id}
- {relation: self, url: '/Patient?_page=1528&_count=15&_elements=id'}
- {relation: next, url: '/Patient?_page=1529&_count=15&_elements=id'}
- {relation: previous, url: '/Patient?_page=1527&_count=15&_elements=id'}
- {relation: last, url: '/Patient?_page=1529&_count=15&_elements=id'}
query-sql: ['SELECT "patient".* FROM "patient" LIMIT ? OFFSET ? ', 15, 22905]
query-time: 31
```

**Comment by ruscoder** [comment]
I think it can be connected to the problem when we specify not-existing page:
```
Get /Patient?_page=100000&_count=20&_elements=id
```

```
resourceType: Bundle
type: searchset
entry: []
total: 0
link:
- {relation: first, url: '/Patient?_page=1&_count=20&_elements=id'}
- {relation: self, url: '/Patient?_page=100000&_count=20&_elements=id'}
query-sql: ['SELECT "patient".* FROM "patient" LIMIT ? OFFSET ? ', 20, 1999980]
query-time: 12
```

**Comment by niquola** [comment]
Thanks, fixed in 0.4.4-SNAPSHOT
